### ISSUE TITLE
error in gnome-shell 3.34

### ISSUE THREAD

**Issue by alirezaimi** [micheleg/dash-to-dock#987]
Hi 
i have this error in gnome-shell 3.34 :

```
Sep 17 19:04:06 lap gnome-shell[1390]: Archive:  /tmp/6U6M8Z.shell-extension.zip
Sep 17 19:04:06 lap gnome-shell[1390]:    creating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/zh_TW/LC_MESSAGES/
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/schemas/org.gnome.shell.extensions.dash-to-dock.gschema.xml
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/gl/LC_MESSAGES/dashtodock.mo
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/Settings.ui
Sep 17 19:04:06 lap gnome-shell[1390]:    creating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/sk/
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/media/logo.svg
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/theming.js
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/pt/LC_MESSAGES/dashtodock.mo
Sep 17 19:04:06 lap gnome-shell[1390]:    creating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/nb/
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/schemas/gschemas.compiled
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/cs/LC_MESSAGES/dashtodock.mo
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/dash.js
Sep 17 19:04:06 lap gnome-shell[1390]:    creating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/zh_CN/
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/id/LC_MESSAGES/dashtodock.mo
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/sr@latin/LC_MESSAGES/dashtodock.mo
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/it/LC_MESSAGES/dashtodock.mo
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/ja/LC_MESSAGES/dashtodock.mo
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/COPYING
Sep 17 19:04:06 lap gnome-shell[1390]:    creating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/nl/LC_MESSAGES/
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/extension.js
Sep 17 19:04:06 lap gnome-shell[1390]:    creating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/tr/
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/docking.js
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/launcherAPI.js
Sep 17 19:04:06 lap gnome-shell[1390]:    creating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/pl/
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/README.md
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/windowPreview.js
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/media/highlight_stacked_bg_h.svg
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/fr/LC_MESSAGES/dashtodock.mo
Sep 17 19:04:06 lap gnome-shell[1390]:    creating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/de/
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/sv/LC_MESSAGES/dashtodock.mo
Sep 17 19:04:06 lap gnome-shell[1390]:    creating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/zh_CN/LC_MESSAGES/
Sep 17 19:04:06 lap gnome-shell[1390]:    creating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/de/LC_MESSAGES/
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/de/LC_MESSAGES/dashtodock.mo
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/intellihide.js
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/media/highlight_stacked_bg.svg
Sep 17 19:04:06 lap gnome-shell[1390]:    creating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/tr/LC_MESSAGES/
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/tr/LC_MESSAGES/dashtodock.mo
Sep 17 19:04:06 lap gnome-shell[1390]:    creating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/sr/
Sep 17 19:04:06 lap gnome-shell[1390]:    creating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/sk/LC_MESSAGES/
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/media/glossy.svg
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/el/LC_MESSAGES/dashtodock.mo
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/utils.js
Sep 17 19:04:06 lap gnome-shell[1390]:    creating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/sr/LC_MESSAGES/
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/nl/LC_MESSAGES/dashtodock.mo
Sep 17 19:04:06 lap gnome-shell[1390]:    creating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/ru/
Sep 17 19:04:06 lap gnome-shell[1390]:    creating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/es/
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/appIconIndicators.js
Sep 17 19:04:06 lap gnome-shell[1390]:    creating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/ru/LC_MESSAGES/
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/hu/LC_MESSAGES/dashtodock.mo
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/zh_CN/LC_MESSAGES/dashtodock.mo
Sep 17 19:04:06 lap gnome-shell[1390]:    creating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/uk_UA/LC_MESSAGES/
Sep 17 19:04:06 lap gnome-shell[1390]:    creating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/nb/LC_MESSAGES/
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/prefs.js
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/uk_UA/LC_MESSAGES/dashtodock.mo
Sep 17 19:04:06 lap gnome-shell[1390]:    creating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/es/LC_MESSAGES/
Sep 17 19:04:06 lap gnome-shell[1390]:    creating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/ar/
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/sk/LC_MESSAGES/dashtodock.mo
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/pl/LC_MESSAGES/dashtodock.mo
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/ru/LC_MESSAGES/dashtodock.mo
Sep 17 19:04:06 lap gnome-shell[1390]:    creating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/ar/LC_MESSAGES/
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/ar/LC_MESSAGES/dashtodock.mo
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/pt_BR/LC_MESSAGES/dashtodock.mo
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/zh_TW/LC_MESSAGES/dashtodock.mo
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/appIcons.js
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/stylesheet.css
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/nb/LC_MESSAGES/dashtodock.mo
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/es/LC_MESSAGES/dashtodock.mo
Sep 17 19:04:06 lap gnome-shell[1390]:   inflating: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/locale/sr/LC_MESSAGES/dashtodock.mo
Sep 17 19:04:06 lap gnome-shell[1390]:  extracting: /home/ali/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/metadata.json
Sep 17 19:04:06 lap gnome-shell[1390]: Extension "dash-to-dock@micxgx.gmail.com" had error: Error: GrabHelper owner must be a Clutter.Actor
Sep 17 19:04:06 lap gnome-software[1478]: State change on user/*/*/shell-extension/dash-to-dock_micxgx.gmail.com/* from available to installed is not OK
Sep 17 19:04:06 lap gnome-software[1478]: dash-to-dock_micxgx.gmail.com has error: Error: GrabHelper owner must be a Clutter.Actor
Sep 17 19:04:06 lap gnome-software[1478]: dash-to-dock_micxgx.gmail.com has error: Error: GrabHelper owner must be a Clutter.Actor

Sep 17 19:04:29 lap gnome-shell[1390]: _clutter_stage_queue_event: assertion 'CLUTTER_IS_STAGE (stage)' failed
Sep 17 19:05:02 lap gnome-shell[1390]: _clutter_stage_queue_event: assertion 'CLUTTER_IS_STAGE (stage)' failed

```

**Comment by mike2307** [comment]
Probably covered by https://github.com/micheleg/dash-to-dock/pull/975

**Comment by mon0theist** [comment]
Also having this issue, new GNOME update breaks Dash to Dock

**Comment by devillove084** [comment]
+1

**Comment by jjgalvez** [comment]
+1

**Comment by AskseL** [comment]
+1

**Comment by Unrasend** [comment]
+1

**Comment by proalexa** [comment]
+1

**Comment by deekosta** [comment]
+1

**Comment by chrisnicholas** [comment]
Has anyone found a workaround/patch for this issue?

**Comment by golopot** [comment]
This seemed to be fixed in master but not yet released. For now you can install the master version by git clone the repo, make, make install, reload gnome.

**Comment by chrisnicholas** [comment]
Excellent, I'll give that a try, thanks!

**Comment by rastersoft** [comment]
Yes, the master branch works fine with Gnome Shell 3.34 and Debian, but still receive a bug message:

Usage of object.actor is deprecated for WorkspaceSwitcherPopup
get@resource:///org/gnome/shell/ui/environment.js:235:29
onScrollEvent@/home/raster/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/docking.js:1269:21

It is triggered when I do a change of workspace using the mouse wheel over the dock.

**Comment by rastersoft** [comment]
Now it is in another line, but still there:

Usage of object.actor is deprecated for WorkspaceSwitcherPopup
get@resource:///org/gnome/shell/ui/environment.js:235:29
onScrollEvent@/home/raster/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/docking.js:1280:21

**Comment by micheleg** [comment]
The warning should be fixed after your PR which is now in master 7ee36193fffe6c1eef7e855b16aa397b4177d76c.

**Comment by arcasinky** [comment]
For what it's worth, I'm seeing it on OpenSuSE 15.2 (Gnome shell 3.34)

```
                                            wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22
                                            _Base.prototype._construct@resource:///org/gnome/gjs/modules/_legacy.js:18:16
                                            newClass@resource:///org/gnome/gjs/modules/_legacy.js:114:32
                                            _init@/home/sylvia/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/docking.js:236:21
                                            wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22
                                            _Base.prototype._construct@resource:///org/gnome/gjs/modules/_legacy.js:18:16
                                            newClass@resource:///org/gnome/gjs/modules/_legacy.js:114:32
                                            _createDocks@/home/sylvia/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/docking.js:1718:20
                                            wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22
                                            _init@/home/sylvia/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/docking.js:1644:9
                                            wrapper@resource:///org/gnome/gjs/modules/_legacy.js:82:22
                                            _Base.prototype._construct@resource:///org/gnome/gjs/modules/_legacy.js:18:16
                                            newClass@resource:///org/gnome/gjs/modules/_legacy.js:114:32
                                            enable@/home/sylvia/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/extension.js:16:19
                                            _callExtensionEnable@resource:///org/gnome/shell/ui/extensionSystem.js:148:13
                                            loadExtension@resource:///org/gnome/shell/ui/extensionSystem.js:280:21
                                            callback@resource:///org/gnome/shell/ui/extensionDownloader.js:232:17
                                            gotExtensionZipFile/<@resource:///org/gnome/shell/ui/extensionDownloader.js:99:13
Aug 11 14:14:00 machine gnome-software[3003]: State change on user/*/GNOME/shell-extension/dash-to-dock_micxgx.gmail.com/* from available to installed is not OK
Aug 11 14:14:00 machine gnome-software[3003]: dash-to-dock_micxgx.gmail.com has error: Error: GrabHelper owner must be a Clutter.Actor
Aug 11 14:14:00 machine gnome-software[3003]: dash-to-dock_micxgx.gmail.com has error: Error: GrabHelper owner must be a Clutter.Actor

```
